internal/engine/upper.go

```diff
@@ -261,6 +261,7 @@ func handleBuildCompleted(ctx context.Context, engineState *store.EngineState, c
 	bs := ms.CurrentBuild
 	bs.Error = err
 	bs.FinishTime = time.Now()
+	bs.DeployID = cb.Result.DeployID
```

Comment:
i'm pretty sure this introduces a race condition in the case where you get the Pod event before the BuildComplete event

one solution to this would be to pick the deploy id in BuildController and attach it to the BuildStarted event, so that it's guaranteed to get into the model before the deploy happens.

Reply:
aaahhhh shoot good call.